moped-editor/src/views/projects/projectView/ProjectView.js

```diff
@@ -148,21 +152,43 @@ const ProjectView = () => {
     ? TABS.findIndex(tab => tab.param === query.get("tab"))
     : 0;
 
+  const DEFAULT_SNACKBAR_STATE = {
+    open: false,
+    message: "Default State",
+    severity: "warning",
+  };
+
   /**
    * @constant {int} activeTab - The number of the active tab
    * @constant {boolean} isEditing - When true, it signals a child component we want to edit the project name
    * @constant {boolean} dialogOpen - When true, the dialog shows
    * @constant {dict} dialogState - Contains the 'title', 'body' and 'actions' as either string or JSX
    * @constant {JSX} anchorElement - The element our 'MoreHorizontal' menu anchors to.
+   * @constant {object} snackbarState - The current state of the snackbar's configuration
    * @constant {boolean} menuOpen - If true, it shows the menu component. Immutable.
    */
   const [activeTab, setActiveTab] = useState(activeTabIndex);
   const [isEditing, setIsEditing] = useState(false);
   const [dialogOpen, setDialogOpen] = useState(false);
   const [dialogState, setDialogState] = useState(null);
   const [anchorElement, setAnchorElement] = useState(null);
+  const [snackbarState, setSnackbarState] = useState(DEFAULT_SNACKBAR_STATE);
   const menuOpen = anchorElement ?? false;
 
+  const handleSnackbarClose = () => {
+    setSnackbarState(DEFAULT_SNACKBAR_STATE);
+  };
+
+  /**
+   * Function which can be passed to child component to allow it to invoke a snackbar which
+   * will persist even after that child component has been deconstructed or otherwise hidden.
+   */
+  const handleSnackbarOpen = snackbarState => {
+    let snackbarStateCopy = {...snackbarState};
+    snackbarStateCopy.open = true;
+    setSnackbarState(snackbarStateCopy);
```

Comment:
looks good. @frankhereford [here's](https://stackoverflow.com/a/40309023/2103784) the best explainer i could find on this. a word of warning with this destructuring approach is that iirc it will only work with shallow objects. if you had a state variable with nested objects you would need to use another method to deep copy.